mypy/build.py

```diff
@@ -1292,13 +1292,7 @@ def find_cache_meta(id: str, path: str, manager: BuildManager) -> CacheMeta | No
     )
 
     # Don't check for path match, that is dealt with in validate_meta().
-    if (
-        m.id != id
-        or m.mtime is None
```

Comment:
Not sure about this. Sure, these attributes may never be None according to their types, but it's good to be defensive against a corrupted cache.

Reply:
I see your point. I think this ultimately points to the annotations for `CacheMeta` being incorrect (or at least less than ideal). But changing those is somewhat of a large task, so I've added a TODO for now.